src/com/untamedears/realisticbiomes/persist/WorldID.java

```diff
@@ -19,27 +24,53 @@ public static void init (Plugin plugin) {
 		
 		World end = plugin.getServer().getWorld("world_the_end");
 		WorldID.endID = end == null ? null : end.getUID();
+		
+		for (World world: plugin.getServer().getWorlds()) {
+			if (!world.getName().equals("world")
+					&& !world.getName().equals("world_nether")
+					&& !world.getName().equals("world_the_end")) {
+				if (otherWorlds == null) {
+					RealisticBiomes.doLog(Level.WARNING, "There are worlds that don't have default names (world, world_nether, world_the_end), experimental feature");
+					otherWorlds = new HashMap<UUID, Integer>();
+				}
+				otherWorlds.put(world.getUID(), world.getUID().hashCode());
+			}
+		}
 	}
 	
 	public static UUID getMCID(int id) {
-		if (id == 0)
+		if (id == 0) {
 			return overworldID;
-		else if (id == 1)
+		} else if (id == 1) {
 			return netherID;
-		else if (id == 2)
+		} else if (id == 2) {
 			return endID;
-		else
+		} else {
+			for (UUID uid: otherWorlds.keySet()) {

```

Comment:
I decided to not to wrap in an `if`, since it will do a similar lookup internally and if it matches it will have done two lookups.